packages/lib-user/src/utils/getBearerToken.js

```diff
@@ -1,4 +1,4 @@
-export default async function getBearerToken (authClient) {
+export async function getBearerToken (authClient) {
```

Comment:
I took another look at this PR by running app-root locally, and visiting https://local.zooniverse.org:3000/users/goplayoutside3/stats. That's my staging username and sometimes I do see stats data, but frequently when I refresh the page, I don't get any data. I narrowed the bug down to this function `getBearerToken()`. If you console.log the `authToken` response from `authClient.checkBearerToken()` while running app-root locally, sometimes it's an empty string. That means UserStats can't fetch my stats data.

The `usePanoptesAuth` hook is a common pattern throughout FEM, so I'm puzzled why it's not working as intended here. I didn't dive too deeply yet, but my next thought process is... if you're relying on the JWToken returned from `auth.checkBearerToken()` as authorization to fetch stats, where/when in app-root or lib-user is the token first set in the browser? Is there a race condition happening where the token doesn't exist yet, but UserStats is trying to get it?

Reply:
In app-root, running `yarn dev`, with the network throttled to Fast 3G - I was also able to reliably recreate the noted issue. 

I think it was race a condition, or maybe relatedly the authClient was firing events, but the useUserStats hook wasn't reacting. In contrast, usePanotpesUser has the change listener, and was updating as expected, but the useUserStats hook was just getting the authClient as a prop and not reacting to any changes, and the refactor further mentioned as follows adds a `auth.checkCurrent()` and awaits the `auth.checkBearerToken()` so it is reacting as expected. I'm not sure and still trying to understand, but that's where my head's at for now.

I refactored useUserStats to get the auth token from panoptes-client/lib/auth similar to how the lib-react-components useUnreadMessages and useUnreadNotifications hooks get the auth token, and it appears to have fixed the issue - lib-user `yarn dev`, app-root `yarn dev`, and app-root `yarn build yarn start` appear to show stats for a staging user and prod user including after page refresh.